packages/vite/src/node/server/index.ts

```diff
@@ -555,7 +562,10 @@ async function startServer(
 
     httpServer.on('error', onError)
 
-    httpServer.listen(port, options.host, () => {
+    const listenHostname = listenPublic
```

Comment:
would this mean you'd have to specify `listenPublic: true` if you want to set a custom hostname?

Reply:
Correct - you would have to explicitly set both.

Reply:
hmm, not sure i like that. If i'm actively setting host it should work without having to set another option too. Imho defaulting host to 127.0.0.1 should suffice. Users can still to `--host 0.0.0.0` to get the old behavior back.

Reply:
not to mention that `listenPublic` may be a misnomer. We don't know the network setup on the machine the devserver is running on. The only safe assumption is that 127.0.0.1 isn't "public".

Reply:
I also think that if a user specifies a `--host`, it is already opting in. 

As @c-bandy was proposing, we should add a message in the console to let the user that.
```
Use --host to expose the server to other devices on this network
```

I think we should implement the same scheme as sirv: https://github.com/lukeed/sirv/tree/master/packages/sirv-cli#network-access
> For security reasons, sirv-cli does not expose your server to the network by default. This means that your machine, and only your machine, will be able to access the localhost server.
> If, however, your coworker wants to access the server from their computer, or you want to preview your work on a mobile device, you must use the --host flag. Only then will your server be accessible to other devices on the same network.
> Using --host without a value is equivalent to --host 0.0.0.0, which is makes it discoverable publicly. You may customize this by passing a different value – but you probably don't need to!

@antfu what do you think? I like the `--host` option without parameters scheme here for the old behavior. And we wouldn't need an extra option